packages/core/test/model.test.ts

```diff
@@ -200,6 +200,31 @@ describe('createModel', () => {
     expect(updatedState.context.age).toEqual(42);
   });
 
+  it('can model actions', () => {
+    const userModel = createModel(
+      {
+        name: 'David',
+        age: 30
+      },
+      {
+        actions: {
+          greet: (message: string) => ({ message })
+        }
+      }
+    );
+
+    createMachine<typeof userModel>({
+      context: userModel.initialContext,
+      initial: 'active',
+      entry: { type: 'greet', message: 'hello' },
+      states: {
+        active: {
+          entry: [userModel.actions.greet('hello')]
```

Comment:
There is IMHO a conceptual clash between this and the other `entry` that is above here. This one is somewhat an equivalent of:
```js
entry: () => greet({ type: 'hello' })
```
while the other one is closer to:
```js
entry: (ctx, ev, { action }) => greet(action)
```

Reply:
Those aren't equivalent; actions that are like `() => something` are not pre-defined actions; they're arbitrary actions.

This is exactly equivalent to the direct object: `{ type: 'greet', message: 'hello' }`

Reply:
What I've meant is that `model.actions.*` are opaque~ to the config etc, it never sees those parameters, while the other one (`{ type: 'greet', message: 'hello' }`) can be introspected. At least in the current `model.events.*`-like form.

It's not obvious how to translate `{ type: 'greet', message: 'hello' }` to `model.actions.greet` because the latter has only positional arguments etc. This is partially what I've meant by those two forms clashing with each other.

Reply:
I'm still confused; this is no different than writing this:

```ts
// same as:
// entry: [model.actions.greet('hello')]
entry: [{ type: 'greet', name: 'hello' }]
```

Reply:
Disregard this whole thread - I've somehow assumed that `model.actions` would actually contain action implementations but they are only used to define action "shaped", the implementations still have to be delivered using regular means (using implementations~ argument).

<img src="https://media.giphy.com/media/l1IBiaH8c1Cxuhq1y/giphy.gif" />